app/jobs/send_webmention_job.rb

```diff
@@ -0,0 +1,88 @@
+class SendWebmentionJob < ApplicationJob
+  queue_as :default
+
+  def endpoint_from_body(html)
+    doc = Nokogiri::HTML(html)
+
+    if !doc.css('[rel~="webmention"]').css("[href]").empty?
+      doc.css('[rel~="webmention"]').css("[href]").attribute("href").value
+    elsif !doc.css('[rel="http://webmention.org/"]').css("[href]").empty?
+      doc.css('[rel="http://webmention.org/"]').css("[href]").attribute("href").value
+    elsif !doc.css('[rel="http://webmention.org"]').css("[href]").empty?
+      doc.css('[rel="http://webmention.org"]').css("[href]").attribute("href").value
+    end
+  end
+
+  def endpoint_from_headers(header)
+    return unless header
+
+    if (matches = header.match(/<([^>]+)>; rel="[^"]*\s?webmention\s?[^"]*"/))
+      matches[1]
+    elsif (matches = header.match(/<([^>]+)>; rel=webmention/))
+      matches[1]
+    elsif (matches = header.match(/rel="[^"]*\s?webmention\s?[^"]*"; <([^>]+)>/))
+      matches[1]
+    elsif (matches = header.match(/rel=webmention; <([^>]+)>/))
+      matches[1]
+    elsif (matches = header.match(/<([^>]+)>; rel="http:\/\/webmention\.org\/?"/))
+      matches[1]
+    elsif (matches = header.match(/rel="http:\/\/webmention\.org\/?"; <([^>]+)>/))
+      matches[1]
+    end
+  end
+
+  # Some pages could return a relative link as their webmention endpoint.
+  # We need to translate this relative link to an absolute one.
+  def uri_to_absolute(uri, req_uri)
+    abs_uri = URI.parse(uri)
+    if abs_uri.host
+      # Already absolute.
+      uri
+    else
+      abs_uri.host = req_uri.host
+      abs_uri.scheme = req_uri.scheme
+      abs_uri.port = req_uri.port
+      abs_uri
+    end
+  end
+
+  def send_webmention(source, target, endpoint)
+    sp = Sponge.new
+    sp.timeout = 10
+    # Don't check SSL certificate here for backward compatibility, security risk
+    # is minimal.
+    sp.ssl_verify = false
+    sp.fetch(endpoint.to_s, :post, {
+      "source" => URI.encode_www_form_component(source),
+      "target" => URI.encode_www_form_component(target)
+    }, nil, {}, 3)
+  end
+
+  def perform(story)
+    # Could have been deleted between creation and now
+    return if story.is_gone?
+    # Need a URL to send the webmention to
+    return if story.url.blank?
+    # Don't try to send webmentions in dev
+    return if Rails.env.development?
+
+    sp = Sponge.new
+    sp.timeout = 10
+    begin
+      response = sp.fetch(URI::RFC2396_PARSER.escape(story.url), :get, nil, nil, {
```

Comment:
third change: as discussed in the other WEBRick-related comments, moves to an escaping util that's already available in stdlib, replacing WEBrick